src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/events/ReferralEventPublisher.kt

```diff
@@ -0,0 +1,19 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events
+
+import org.springframework.context.ApplicationEvent
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Referral
+
+enum class ReferralEventType {
+  SENT
+}
+
+class ReferralEvent(source: Any, val type: ReferralEventType, val referral: Referral) : ApplicationEvent(source)
```

Comment:
Do we need the `type`? We only have one event and we could name it `ReferralSentEvent(...)`

Reply:
the listener has to be scoped to the event class, so having each event as a seperate class means a seperate listener for each event, which just means more boilerplate code when adding new events, instead of just adding a new type to the list.

Reply:
🤔 Thinking more about this, two further things popped up

- I'm not sure if we will have anything else as "referral event" as the subsequent ones will be "related" to the referral but happen about "action plan", "appointment", "end of service report"
- We don't know how different listeners have to be in the future -- it doesn't sound like we can yet make a call on what the reasonable categorisation might be. (Branching should be built in when there's more than 1 branch)

I think it's better to revisit this when we have 2 separate events.